themes/default/content/docs/intro/concepts/resources/functions.md

```diff
@@ -0,0 +1,291 @@
+</div>
+<div>
+<pulumi-choosable type="language" values="yaml">
+
+```yaml
+variables:
+  example:
+    Fn::Invoke:
+      Function: aws:apigateway:getDomainName
+      Arguments:
+        domainName: api.example.com
+```
+
+</pulumi-choosable>
+</div>
+</pulumi-examples></div>
+
+Provider functions are exposed in each language as a regular functions, in two variations:
+
+ 1. a function that accepts plain arguments (strings and so on) and returns a Promise, or blocks until the result is available; and,
+ 2. a function that accepts `Input` values and returns an [Output]({{< relref "/docs/intro/concepts/inputs-outputs" >}}).
+
+The documentation for a provider function will tell you the name and signature for each of the variations.
+
+### Invoke options
+
+Each function also accepts "invoke options", either as an object or as varargs depending on the host
+language. The options are as follows:
+
+| Option | Explanation                                                  |
+|--------|--------------------------------------------------------------|
+| parent | Supply a parent resource, which will be used to determine default providers, and to calculate the protect property. |
+| provider | Supply the provider to use explicitly. |
+| version | Use exactly this version of the provider plugin. |
+| pluginDownloadURL | Download the provider plugin from this URL. The download URL is otherwise inferred from the provider package. |
+| _async_ | _This option is deprecated and will be removed in a future release_ |
+
+The `parent` option has a similar purpose to the [parent option]({{< relref "/docs/intro/concepts/resources/options/parent" >}}) used when creating a resource. The parent is consulted when determining the provider to use, and the [`protect`]({{< relref "/docs/intro/concepts/resources/options/protect" >}}) status of any resources created as a result of the function invocation.
+
+The `provider` option gives an explicit provider to use when running the invoked function. This is useful, for example, if you want to invoke a function in each of a set of AWS regions.
+
+The `version` option specifies an exact version for the provider plugin. This can be used when you need to pin to a specific version to avoid a backward-incompatible change.
+
+The `pluginDownloadURL` option gives a URL for fetching the provider plugin. It may be necessary to supply this for third-party packages (those not hosted at [https://get.pulumi.com](https://get.pulumi.com)).
+
+## Getter functions
+
+A special case of functions is the static `get` function, which is available on all resource types, to look up an existing resource’s ID. The `get` function is different from the `import` function. The difference is that, although the resulting resource object’s state will match the live state from an existing environment, the resource will not be managed by Pulumi. A resource read with the `get` function will never be updated or deleted by Pulumi during an update.
```

Comment:
I'm not sure we should call this a special case of package functions. This is really a wholly different thing. 

Most deeply, while "functions" are just executed and return values - the .get methods actually lead to a resource operation which is recorded in the state file and in the progress display. 

For example, this mean that you can put try/catch around functions, but not around .get (because the latter failing is a resource operation failing in the engine, which will fail the deployment).

The .get calls are a lot more like a `new Res()` call in how they operate - with the key difference that they never create a resource, they always look it up based on id.

Reply:
Okay yes, I took the spirit of the original issue as being "both of these are functions and should be discussed together". But as you've explained it just above, it sounds more like they are different enough to each deserve their own treatment, as separate pages.

Reply:
Putting them on the same page will imply that they're mostly the same, even if I remove the offending claim :-) If there's consequences for an end user to mistakenly assuming they are the same, I'd prefer to keep them on separate pages.

Reply:
I rearranged things and gave functions and methods their own page -- they are more similar to each other than to getter functions. I left the latter page as it is.